pkg/build/controller/image_change_controller.go

```diff
@@ -81,16 +80,23 @@ func (c *ImageChangeController) HandleImageRepo(repo *imageapi.ImageRepository)
 
 		if shouldBuild {
 			glog.V(4).Infof("Running build for buildConfig %s in namespace %s", config.Name, config.Namespace)
-			b := buildutil.GenerateBuildFromConfig(config, nil, subs)
-			if err := c.BuildCreator.Create(config.Namespace, b); err != nil {
-				return fmt.Errorf("error starting build for buildConfig %s: %v", config.Name, err)
-			}
+			// first update the config
 			if err := c.BuildConfigUpdater.Update(config); err != nil {
-				// This is not a retryable error because the build has been created.  The worst case
-				// outcome of not updating the buildconfig is that we might rerun a build for the
-				// same "new" imageid change in the future, which is better than running the build
-				// 2+ times by retrying it here.
-				return ImageChangeControllerFatalError{Reason: fmt.Sprintf("Error updating buildConfig %s with new LastTriggeredImageID", config.Name), Err: err}
+				// This is not a retryable error. The worst case outcome of not updating the buildconfig
+				// is that we might rerun a build for the same "new" imageid change in the future,
+				// which is better than guaranteeing we run the build 2+ times by retrying it here.
+				return ImageChangeControllerFatalError{
+					Reason: fmt.Sprintf("Error updating buildConfig %s with new LastTriggeredImageID", config.Name),
+					Err:    err,
+				}
+			}
+			// and then trigger the generation for that config
+			request := &buildapi.BuildRequest{BuildConfigName: config.Name}
+			if _, err := c.BuildGenerator.Generate(config.Namespace, request); err != nil {
+				return ImageChangeControllerFatalError{
```

Comment:
this is what i was trying to say on the scrum...  this did not used to be a fatal error, it used to be that
1) GenerateBuildFromConfig never returned an error
2) if the actual Create build REST call failed, we returned a retryable error

Now you've got it so that if Generate (which is the combination of GenerateFromBuildConfig and the Create call) fails, you throw a fatal error so it won't be retried.